### PR TITLE
Add Python syntax highlighting to `README.rst`

### PR DESCRIPTION
**Pull request by Benjamin-Loison**
Verified exhaustiveness with:

```bash
find -name '*.rst'
```

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -301,8 +307,9 @@ thing on the IPython notebook it scales up to 44MB.
 
 If you'd like to get the memory consumption of a Python function, then
 you should specify the function and its arguments in the tuple ``(f,
-args, kw)``. For example::
+args, kw)``. For example:
 
+.. code-block:: python
```
Comment (altendky) [inline README.rst:312]:
```suggestion
.. code-block:: python-console
```

File: README.rst [hunk 2]
```diff
@@ -325,6 +332,8 @@ REPORTING
 The output can be redirected to a log file by passing IO stream as
 parameter to the decorator like @profile(stream=fp)
 
+.. code-block:: python
```
Comment (altendky) [inline README.rst:335]:
```suggestion
.. code-block:: python-console
```

File: README.rst [hunk 3]
```diff
@@ -343,6 +352,8 @@ when we need to use RotatingFileHandler.
 The output can be redirected to logger module by simply making use of
 LogFile of memory profiler module.
 
+.. code-block:: python
```
Comment (altendky) [inline README.rst:355]:
```suggestion
.. code-block:: python-console
```

File: README.rst [hunk 4]
```diff
@@ -354,11 +365,13 @@ could be cumbersome and one can choose only entries with increments
 by passing True to reportIncrementFlag, where reportIncrementFlag is
 a parameter to LogFile class of memory profiler module.
 
+.. code-block:: python
```
Comment (altendky) [inline README.rst:368]:
```suggestion
.. code-block:: python-console
```

File: README.rst [hunk 5]
```diff
@@ -441,6 +466,8 @@ Currently, the backend can be set via the CLI
 
 and is exposed by the API
 
+.. code-block:: python
```
Comment (altendky) [inline README.rst:469]:
```suggestion
.. code-block:: python-console
```

File: README.rst [hunk 6]
```diff
@@ -282,6 +287,7 @@ necessarily a Python program), a string containing some python code to
 be evaluated or a tuple ``(f, args, kw)`` containing a function and its
 arguments to be evaluated as ``f(*args, **kw)``. For example,
 
+.. code-block:: python
```
Comment (altendky) [inline README.rst:290]:
```suggestion
.. code-block:: python-console
```

**Review by altendky — COMMENTED** [review]
Thanks for an easy review!  And the improvements.  `:]`  What do you think of using `python-console` for those blocks as in the suggestions?

**Comment by Benjamin-Loison** [comment]
Thank you for suggesting using Python console syntax highlighting, I like the idea of using it to be more precise. However, as [this code snippet](https://github.com/Benjamin-Loison/memory_profiler/blob/864e9f4d661b0680a892ef272d7d30894469a168/README.rst?plain=1#L316-L320) shows in the `Preview` tab, it has a syntax highlighting for the indented code (by precising Python syntax highlighting), but [not when using Python console syntax highlighting](https://github.com/Benjamin-Loison/memory_profiler/blob/python_console_syntax_highlighting/README.rst?plain=1#L316-L320). Without even talking about the rendering in the `Code` tab which *doesn't work at all for Python console syntax highlighting.

So it Python console GitHub rendering doesn't seem perfect yet, hence I don't have any strong opinion and I let you choose the one you prefer.

**Review by altendky — APPROVED** [review]
Always something, isn't it...  Oh well.

**Pull request merged**